# TEST -- PYTHON USE CASE 3

**Dataset** : Titanic dataset. 

You can download it here. [Titanic.csv](https://drive.google.com/open?id=15IyT1ODuDKgZb8WN6iG64hFJRZWJTSAz)

**Background** : The dataset contains data for 887 of the real Titanic passengers. Each row represents one person. The columns describe different attributes about the person including whether they survived (S), their age (A), their passenger-class (C), their sex (G) and the fare they paid (X). The table below shows the Data Dictionary.

**Objective** : Create a python script that contains the script to answer the questions below. Please put comments on the script to show which line of codes answers which questions. Please upload the script after you finish.

# Import Library

In [15]:
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

# Import dataset

In [4]:
df = pd.read_csv('titanic.csv')

In [5]:
df.isnull().sum().sort_values(ascending = False)

Fare                       0
Parents/Children Aboard    0
Siblings/Spouses Aboard    0
Age                        0
Sex                        0
Name                       0
Pclass                     0
Survived                   0
dtype: int64

# I. What is the dimension (col, row) of the data frame?

In [6]:
df.shape

(887, 8)

# II. How to know data type of each variable?

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
Survived                   887 non-null int64
Pclass                     887 non-null int64
Name                       887 non-null object
Sex                        887 non-null object
Age                        887 non-null float64
Siblings/Spouses Aboard    887 non-null int64
Parents/Children Aboard    887 non-null int64
Fare                       887 non-null float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.5+ KB


# III. How many passengers survived (Survived=1) and not-survived (Survived=0)?

In [31]:
# III
a = df['Survived'].value_counts()
print("survived : ",a[1])
print("Not-survived : ",a[0])

survived :  342
Not-survived :  545


# IV. How to drop column ‘Name’ from the data frame?

In [9]:
b = df.drop(['Name'], axis=1)
b.head(3)

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   

   Parents/Children Aboard     Fare  
0                        0   7.2500  
1                        0  71.2833  
2                        0   7.9250

# V. Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)

In [10]:
df['family'] = df['Siblings/Spouses Aboard'] + df['Parents/Children Aboard']
df.head(3)

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  \
0    male  22.0                        1                        0   7.2500   
1  female  38.0                        1                        0  71.2833   
2  female  26.0                        0                        0   7.9250   

   family  
0       1  
1       1  
2       0

# VI. As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for non-missing value).

In [16]:
df['Age_miss']='null'
df['Age_miss'][df['Age'] == None] = 'Yes'
df['Age_miss'][df['Age'] != None] = 'No'
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  \
0    male  22.0                        1                        0   7.2500   
1  female  38.0                        1                        0  71.2833   
2  female  26.0                        0                        0   7.9250   
3  female  35.0                        1                        0  53.1000   
4    male  35.0                        0                        0   8.0500   

   family Age_miss  
0       1       No  
1       1       No  
2       0       No  
3       1       No  
4       0       No

# VII. Please fill Age missing value with means of existing Age values

In [18]:
df['Age'].fillna(df['Age'].mean())

0      22.0
1      38.0
2      26.0
3      35.0
4      35.0
5      27.0
6      54.0
7       2.0
8      27.0
9      14.0
10      4.0
11     58.0
12     20.0
13     39.0
14     14.0
15     55.0
16      2.0
17     23.0
18     31.0
19     22.0
20     35.0
21     34.0
22     15.0
23     28.0
24      8.0
25     38.0
26     26.0
27     19.0
28     24.0
29     23.0
       ... 
857    21.0
858    48.0
859    14.0
860    24.0
861    42.0
862    27.0
863    31.0
864    23.0
865     4.0
866    26.0
867    47.0
868    33.0
869    47.0
870    28.0
871    15.0
872    20.0
873    19.0
874    23.0
875    56.0
876    25.0
877    33.0
878    22.0
879    28.0
880    25.0
881    39.0
882    27.0
883    19.0
884     7.0
885    26.0
886    32.0
Name: Age, Length: 887, dtype: float64

# VIII. What is the maximum passenger Age who survived from the tragedy?

In [19]:
sur = df.loc[(df.Survived == 1)]
maxx = sur.sort_values('Age', ascending = False)
print ('Max Age Survived : ', maxx['Age'].head(1).tolist())

Max Age Survived :  [80.0]


# IX. How many passengers survived from each ‘PClass’?

In [25]:
sur = df.loc[(df.Survived == 1)]
pgrup = sur.groupby('Pclass')
p_cnt = pgrup['Pclass'].count()
print('Pclass 1 survived : ',p_cnt[1])
print('Pclass 2 survived : ',p_cnt[2])
print('Pclass 3 survived : ',p_cnt[3])

Pclass 1 survived :  136
Pclass 2 survived :  87
Pclass 3 survived :  119


# X. How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for tttanic1 and 0.3 for titanic2 ?

In [30]:
from sklearn.model_selection import train_test_split
titanic1, titanic2 = train_test_split(df, test_size=0.3)
print('Total titanic1 :', titanic1.shape[0])
print('Total titanic2 :', titanic2.shape[0])

Total titanic1 : 620
Total titanic2 : 267


# Conclussion
Tujuan dari case ini adalah untuk menjawab pertanyaan-pertanyaan yang di minta. jawaban dari setiap pertanyaan itu ada pada semua kode program diatas.
<br>Informasi-informasi yang didapatkan antara lain :
<br>**Penumpang selamat** :  342
<br>**Penumpang tidak selamat** :  545
<br>**Penumpang selamat pada Pclass 1 survived** :  136
<br>**Penumpang selamat pada Pclass 2 survived** :  87
<br>**Penumpang selamat pada Pclass 3 survived** :  119